src/Magnum/Octree/Octree.h

```diff
@@ -0,0 +1,274 @@
+         */
+        bool isLeafNode() const {
+            return _isLeafNode;
+        }
+
+        /**
+         * @brief Set if octree is a leaf node.
+         *
+         * An octree is a leaf node if it has no children.
+         */
+        void setLeafNode(bool isLeafNode) {
+            _isLeafNode = isLeafNode;
+        }
+
+        /** @brief Whether this node is empty */
+        bool empty() const {
+            return _data.size() == 0;
+        }
+
+        /**
+         * @brief Build the octree for points and a maximal depth
+         * @param points      The points that should get stored in the octree
+         * @param maxDepth    The maximal depth of the octree
+         * @return Reference to self (for method chaining)
+         */
+        Octree<T>& build(const std::vector<Range3D>& entries, const std::vector<T>& data, const Int maxDepth) {
+            CORRADE_ASSERT(data.size() == entries.size(), "Entries and data need to be same size.", *this);
+
+            auto entryItor = entries.begin();
+            auto entryEnd = entries.end();
+            auto dataItor = data.begin();
+
+            createInitialBounds(entries, maxDepth);
+
+            while(entryItor != entryEnd) {
+                insert(*entryItor, *dataItor);
+                entryItor++;
+                dataItor++;
+            }
+
+            return *this;
+        }
+
+        /**
+         * @brief Get the child of the octree in which the point is stored, if the point exists
+         * @param point    The point that should be found
+         * @return The index of the child potentially containing the point
+         */
+        Int getOctantContainingBox(const Range3D& point) const;
+
+        /**
+         * @brief Get all points contained of the octree within given bounds
+         * @param found    All the points found in the specified bounds
+         * @param min      The minimal corner of the search bounds
+         * @param max      The maximal corner of the search bounds
+         * @return Reference to self (for method chaining)
+         */
+        Octree<T>& points(std::vector<T>& resultData, const Vector3& min, const Vector3& max);
+        Octree<T>& points(std::vector<T>& resultData, const Vector4 (&frustum)[6]);
+        Octree<T>& points(std::vector<T>& resultData);
+
+    private:
+        /**
+         * @brief Private constructor for setting the center, radius and depth of the octree manually
+         * @param center    The center of the octree
+         * @param radius    The radius of the octree
+         * @param depth     The depth of the octree
+         */
+        Octree(Vector3 center, Float radius, Int depth, bool isLeafNode) :
+            _center { center },
+            _radius(radius),
+            _depth(depth),
+            _isLeafNode(isLeafNode)
+        {
+            for(int i = 0; i < 8; ++i) {
+                _children[i] = nullptr;
+            }

```

Comment:
Similarly to the above, just `_children{}` in the constructor initializer list (or in-class).